Added preview feature for uploaded images

iamhks opened the pull request:
## Summary

Description of PR here...
Closes #164 

Screenshots:
<img width="464" alt="Screenshot 2023-03-21 at 01 09 49" src="https://user-images.githubusercontent.com/43777817/226449200-cb873c99-7eb9-480d-8769-f5ad7e03717f.png">
<img width="519" alt="Screenshot 2023-03-21 at 01 09 29" src="https://user-images.githubusercontent.com/43777817/226449246-5bc8e894-fd79-4700-91af-34633874c45b.png">

iamhks committed:
Added preview feature for uploaded images

tuxology commented:
Thanks for this @iamhks 👍 Can we reuse some parts from the image preview work done already by @souad988  here? https://github.com/unstructuredstudio/zubhub/blob/master/zubhub_frontend/zubhub/src/components/upload_file/uploadFileScripts.js#L96-L116

tuxology commented on zubhub_frontend/zubhub/public/index.html:
```diff
@@ -47,6 +47,7 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
     <title>Zubhub</title>
+    <script src="https://kit.fontawesome.com/58aa8d2b83.js" crossorigin="anonymous"></script>
```
It would be nice if we can reuse some earlier and work and avoid adding new FA js

iamhks replied:
Updated the PR

iamhks committed:
review changes

iamhks commented:
@tuxology I have updated the PR to show images just like create activity screen.

Although what I said [here](https://github.com/unstructuredstudio/zubhub/issues/164#issuecomment-1476843811) is still valid and as expected the pictures show up just like expected:
<img width="522" alt="Screenshot 2023-03-25 at 03 54 44" src="https://user-images.githubusercontent.com/43777817/227704658-42a24c14-da92-4da7-ae44-dcf2758f6349.png">

But I also think it is nice to have the same consistent design throughout the site so preview Image in createProject replicates createActivity's 2nd screen. 
I have also updated the caption to show dynamic number of images as per addition/removal like here:
<img width="416" alt="Screenshot 2023-03-25 at 13 12 18" src="https://user-images.githubusercontent.com/43777817/227704764-d46407ae-7b7a-4dd4-b82e-db834363c44c.png">

Pls review at your convenience.

tuxology commented:
@iamhks Thanks, this is shaping up nice! I'll be testing this, but in the meantime one final nimor thing: can we use the close buttons same as what @psalmuelle used in his PR https://github.com/unstructuredstudio/zubhub/pull/613 ? This will allow us to have consistent closing buttons. This way we have same closing buttons everywhere. We can also use what you have but just ensure the background is white and not transparent for the close buttons. 

Please also test this for responsiveness and share screenshots if possible 🙏

iamhks committed:
Icon button changes

iamhks commented:
Sure @tuxology, I have made the changes and here are the screenshots:

<img width="723" alt="Screenshot 2023-03-26 at 09 13 55" src="https://user-images.githubusercontent.com/43777817/227754342-3de08d98-bf43-456e-a27f-165497a9dd50.png">

<img width="560" alt="Screenshot 2023-03-26 at 09 12 54" src="https://user-images.githubusercontent.com/43777817/227754344-49f02f4d-69c0-499c-8f7e-60ce51be4683.png">

<img width="678" alt="Screenshot 2023-03-26 at 09 14 39" src="https://user-images.githubusercontent.com/43777817/227754334-756e0c27-84d6-4d63-9802-e9e7545a8ee5.png">

<img width="1440" alt="Screenshot 2023-03-26 at 09 15 09" src="https://user-images.githubusercontent.com/43777817/227754337-f4077211-816f-433c-8ac4-cc497bbcd404.png">

iamhks commented:
@tuxology Are we good with this PR to merge?

iamhks commented:
Hi @tuxology Any update on this?

tuxology commented:
Done as part of #718 Closing this.
